batch.go

```diff
@@ -181,133 +182,177 @@ func Batch(cc *cli.Context) error {
 		return fmt.Errorf("failed to read input directory: %w", err)
 	}
 
-	grp, ctx := errgroup.WithContext(ctx)
-	grp.SetLimit(batchOpts.concurrency)
+	for _, fname := range fnames {
+		fcontent, err := fs.ReadFile(infs, fname)
+		if err != nil {
+			return fmt.Errorf("failed to read processing profile %q: %w", fname, err)
+		}
+
+		var profile ProcessingProfile
+		if err := yaml.Unmarshal(fcontent, &profile); err != nil {
+			return fmt.Errorf("failed to unmarshal processing profile: %w", err)
+		}
+		profile.Dir = filepath.Join(batchOpts.inDir, profile.Dir)
+		if len(profile.Variants) == 0 {
+			profile.Variants = []map[string]any{{}}
+		}
+
+		if err := profile.processPlotDefs(ctx, cfg); err != nil {
+			return fmt.Errorf("processing plot definitions: %w", err)
+		}
+	}
+
+	return nil
+}
 
-	org := new(Organizer)
-	absOutDir, err := filepath.Abs(batchOpts.outDir)
+func (p *ProcessingProfile) processPlotDefs(ctx context.Context, cfg *PlotConfig) error {
```

Comment:
I've attached this method to the `ProcessingProfile` struct.  Not sure about this 🤷‍♂️ made it easy to access a few parameters